python/lsst/meas/base/plugins.py

```diff
@@ -71,6 +73,72 @@
                     TransformClass=bl.ApertureFluxTransform, executionOrder=BasePlugin.FLUX_ORDER)
 
 # --- Single-Frame Measurement Plugins ---
+class SingleFrameFocalPlaneConfig(SingleFramePluginConfig):
+    pass
+
+@register("base_FocalPlane")
+class SingleFrameFocalPlanePlugin(SingleFramePlugin):
+    '''
+    Algorithm to calculate position on focal plane
+    '''
+
+    ConfigClass = SingleFrameFocalPlaneConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.APCORR_ORDER
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+        self.focalValue = lsst.afw.table.Point2DKey.addFields(schema, name, "Position on the focal plane",
+                "mm")
+        self.focalFlag = schema.addField(name + "_flag", type="Flag", doc="Set to 1 for any fatal Failure")
+
+    def measure(self, measRecord, exposure):
+        center = measRecord.getCentroid()
+        det = exposure.getDetector()
+        if not det:
```

Comment:
If you don't have a detector, you don't need a centroid. Why not check for the former first, then not bother fetching the centroid if you're not going to use it?